packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -33,6 +44,72 @@ export class ForceSourceDeploySourcePathExecutor extends BaseDeployExecutor {
     return commandBuilder.build();
   }
 
+  public async execute(response: ContinueResponse<string>): Promise<void> {
+    const betaDeployRetrieve = sfdxCoreSettings.getBetaDeployRetrieve();
+    // this supported types logic is temporary until we have a way of generating the metadata type from the path
+    // once we have the metadata type we can check to see if it is a toolingsupportedtype from that util
+    const supportedType =
+      path.extname(response.data) === APEX_CLASS_EXTENSION ||
+      response.data.includes(`${APEX_CLASS_EXTENSION}-meta.xml`);
+    const multipleSourcePaths = response.data.includes(',');
+
+    if (betaDeployRetrieve && supportedType && !multipleSourcePaths) {
+      const startTime = process.hrtime();
```

Comment:
I would recommend breaking some of this logic out into a few methods. I think it would make it easier to read and unit test.

Reply:
Yeah, I think this will happen once we start using the new library and introduce a new executor for non-cli work. Those things should start happening fairly soon.